### PR TITLE
Add PV.Name to volume tags.

### PR DESCRIPTION
**Pull request by jsafrane**
We add claim.Name and claim.Namespace as tags to AWS EBS / GCE PD / OpenStack
Cinder volumes created by Kubernetes. To easily match Kubernetes volumes and
cloud volumes, let's add also PV.Name.

This allows easier matching between Kubernetes and real cloud volumes.

### CODE REVIEW

**Comment by jsafrane** [comment]
cc: @kubernetes/rh-storage

File: pkg/controller/persistentvolume/persistentvolume_provisioner_controller.go [hunk 1]
```diff
@@ -330,15 +330,19 @@ func (controller *PersistentVolumeProvisionerController) Stop() {
 	}
 }
 
-func newProvisioner(plugin volume.ProvisionableVolumePlugin, claim *api.PersistentVolumeClaim) (volume.Provisioner, error) {
+func newProvisioner(plugin volume.ProvisionableVolumePlugin, claim *api.PersistentVolumeClaim, pv *api.PersistentVolume) (volume.Provisioner, error) {
+	tags := make(map[string]string)
+	tags[cloudVolumeCreatedForClaimNamespaceTag] = claim.Namespace
+	tags[cloudVolumeCreatedForClaimNameTag] = claim.Name
+	if pv != nil {
```
Comment (markturansky) [inline pkg/controller/persistentvolume/persistentvolume_provisioner_controller.go]: When would PV not be nil?  We first make the PV from the template and persist it with ClaimRef to the PVC.

It seems like nil PV here should be an error.

Comment (jsafrane) [inline pkg/controller/persistentvolume/persistentvolume_provisioner_controller.go]: `newProvisioner` is called twice and at the first call the PV is not yet availabe, it's created by the returned provisioner.

https://github.com/kubernetes/kubernetes/pull/18957/files#diff-fc923807bec23504369a29c0f94b6055R175

Comment (derekwaynecarr) [inline pkg/controller/persistentvolume/persistentvolume_provisioner_controller.go]: can you add a comment to that effect?  I am less familiar with this area when doing the review and had a similar thought.

Comment (jsafrane) [inline pkg/controller/persistentvolume/persistentvolume_provisioner_controller.go]: I added a short note.

**Comment by derekwaynecarr** [comment]
Is there no test case that can verify the tags on a pvc?  Can we add a mock unit test?

**Comment by jsafrane** [comment]
> Is there no test case that can verify the tags on a pvc? Can we add a mock unit test?

I'll try to squeeze it into some provisioning test.

**Comment by jsafrane** [comment]
Test added.

File: pkg/controller/persistentvolume/types.go [hunk 2]
```diff
@@ -32,10 +32,13 @@ const (
 	qosProvisioningKey = "volume.alpha.kubernetes.io/storage-class"
 	// Name of a tag attached to a real volume in cloud (e.g. AWS EBS or GCE PD)
 	// with namespace of a persistent volume claim used to create this volume.
-	cloudVolumeCreatedForNamespaceTag = "kubernetes.io/created-for/pvc/namespace"
+	cloudVolumeCreatedForClaimNamespaceTag = "kubernetes.io/created-for/pvc/namespace"
```
Comment (thockin) [inline pkg/controller/persistentvolume/types.go]: This should fail validation - only one / is allowed

Comment (thockin) [inline pkg/controller/persistentvolume/types.go]: Sorry, I interpreted this as a k8s label, not a cloud tag

**Comment by thockin** [comment]
add to whitelist

**Comment by markturansky** [comment]
@k8s-bot test this

**Comment by markturansky** [comment]
```

!!! [0113 17:42:19] Timed out waiting for apiserver:  to answer at http://127.0.0.1:8080/healthz; tried 120 waiting 1 between each
!!! Error in ./hack/test-update-storage-objects.sh:48
  'return 1' exited with status 1
Call stack:
```

**Comment by jsafrane** [comment]
@k8s-bot test this

**Comment by jsafrane** [comment]
All green. @derekwaynecarr , thanks for review.

**Pull request merged**